src/main/java/net/snowflake/client/jdbc/SnowflakeChunkDownloader.java

```diff
@@ -675,52 +675,56 @@ private void logOutOfMemoryError() {
   @Override
   public DownloaderMetrics terminate() throws InterruptedException {
     if (!terminated.getAndSet(true)) {
-      if (executor != null) {
```

Comment:
Not much has changed EXCEPT the addition of a try/finally block. The "try" contains all non-memory-freeing code, including the line **executor.awaitTermination()** which has the possibility of throwing an InterruptedException.

The finally block contains the logic to free the chunk data.

Now, when an InterruptedException is thrown, the memory is freed prior to driver execution ending.